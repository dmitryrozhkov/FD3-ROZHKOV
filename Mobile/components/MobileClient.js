import React from 'react';
import PropTypes from 'prop-types';

import './MobileClient.css';

class MobileClient extends React.PureComponent {

  static propTypes = {
    info:PropTypes.shape({
      id: PropTypes.number.isRequired,
      fio: PropTypes.string.isRequired,
      balance: PropTypes.number.isRequired,
    }),
  };

  state = {
    info: this.props.info,
  };

  componentWillReceiveProps = (newProps) => {    
    this.setState({info:newProps.info});
  };

  render() {

    console.log("MobileClient id="+this.state.info.id+" render");
    let countColor = this.props.color?"red":"black";    
    return (
      <div className='MobileClient'>
        <span className='MobileClientBalance' style={{color:countColor}}>{this.state.info.balance}</span>
        <span className='MobileClientFIO' style={{color:countColor}}>{this.state.info.fio}</span>        
      </div>
    );

  }

}

export default MobileClient;

 
