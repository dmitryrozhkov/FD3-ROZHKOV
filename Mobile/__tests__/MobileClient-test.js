"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import MobileClient from '../components/MobileClient';

test('работа MobileClient', () => {

  let clientsArr= [
    {id:101, fio:"Иванов И.И.", balance:200}, 
    {id:105, fio:"Сидоров С.С.", balance:-250}, 
    {id:110, fio:"Петров П.П.", balance:-180},
    {id:120, fio:"Григорьев Г.Г.", balance:220},
  ]

  clientsArr.forEach( (client) =>     {     
    
  const component = renderer.create(
    <MobileClient color = "black" key={client.id} info={client} />
  );

  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
  }); 
    
});