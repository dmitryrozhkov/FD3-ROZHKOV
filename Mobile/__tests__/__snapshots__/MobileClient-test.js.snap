// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа MobileClient 1`] = `
<div
  className="MobileClient"
>
  <span
    className="MobileClientBalance"
  >
    200
  </span>
  <span
    className="MobileClientFIO"
  >
    Иванов И.И.
  </span>
</div>
`;

exports[`работа MobileClient 2`] = `
<div
  className="MobileClient"
>
  <span
    className="MobileClientBalance"
  >
    -250
  </span>
  <span
    className="MobileClientFIO"
  >
    Сидоров С.С.
  </span>
</div>
`;

exports[`работа MobileClient 3`] = `
<div
  className="MobileClient"
>
  <span
    className="MobileClientBalance"
  >
    -180
  </span>
  <span
    className="MobileClientFIO"
  >
    Петров П.П.
  </span>
</div>
`;

exports[`работа MobileClient 4`] = `
<div
  className="MobileClient"
>
  <span
    className="MobileClientBalance"
  >
    220
  </span>
  <span
    className="MobileClientFIO"
  >
    Григорьев Г.Г.
  </span>
</div>
`;
